src/compiler/checker.ts

```diff
@@ -263,6 +263,30 @@ namespace ts {
         Uncapitalize: IntrinsicTypeKind.Uncapitalize
     }));
 
+    enum TypeChecks {

```

Comment:
Instead of returning a succeed/fail type with an error inside, I'd keep the `reportError` parameter and associated code. Then just have `isPropertyAccessible` always pass `false`.

We [almost never?] use wrapper types like this in the compiler because they're less efficient than using primitives -- something like `DiagnosticMessage | undefined` is the usual way. That requires a name change to be readable: probably `getPropertyAccessibilityError`.